tools/update.py

```diff
@@ -65,65 +86,74 @@ def main(options, args):
     log("Updating import-statements in %d files..." % len(registry.keys()))
 
     if options.dry_run:
-        queue = ['patch', '-p0', '--dry-run'],
+        queue = (["patch", "-p0", "--dry-run"],)
     else:
-        queue = ['patch', '-p0', '--dry-run'], ['patch', '-p0']
+        queue = ["patch", "-p0", "--dry-run"], ["patch", "-p0"]
 
     for args in queue:
         if options.verbose:
-            args.append('--verbose')
+            args.append("--verbose")
         else:
-            args.append('--quiet')
+            args.append("--quiet")
 
         proc = subprocess.Popen(
             " ".join(args),
             shell=True,
             stdin=subprocess.PIPE,
             stdout=subprocess.PIPE,
-            )
+        )
 
         diffs = []
         for filename, body in registry.iteritems():
             if options.verbose:
-                changes = body.count('+++ ')
-                status("%s (%d %s)." % (
-                    filename, changes, 'changes' if (changes > 1) else 'change'))
+                changes = body.count("+++ ")
+                status(
+                    "%s (%d %s)."
+                    % (
+                        filename,
+                        changes,
+                        "changes" if (changes > 1) else "change",
+                    )
+                )
 
             diffs.append(body)
 
         if not diffs:
             break
 
-        output, err = proc.communicate("\n".join(diffs)+'\n')
-        
+        output, err = proc.communicate("\n".join(diffs) + "\n")
+
         if proc.returncode != 0:
-            map(status, output.split('\n'))
+            map(status, output.split("\n"))
```

Comment:
This is a bug on Python 3: map is now a generator and thus `status()` will not be called since nothing loops over `map`'s return value.

Replace with a for loop.

Reply:
Good catch!